linkerd/meshtls/rustls/src/client.rs

```diff
@@ -91,9 +99,21 @@ where
     }
 
     fn call(&mut self, io: I) -> Self::Future {
-        tokio_rustls::TlsConnector::from(self.config.clone())
-            .connect(self.server_name.clone(), io)
-            .map_ok(ClientIo)
+        let server_id = self.server_id.clone();
+        Box::pin(
+            tokio_rustls::TlsConnector::from(self.config.clone())
+                // XXX(eliza): it's a bummer that the server name has to be cloned here...
+                .connect(self.server_name.clone(), io)
+                .map(move |s| match s {
+                    Ok(s) => {
+                        let (_, conn) = s.get_ref();
+                        let end_cert = extract_cert(conn)?;
+                        verify::verify_id(end_cert, &server_id)?;
+                        Ok(ClientIo(s))
+                    }
+                    Err(err) => Err(err),
+                }),
```

Comment:
nit, take it or leave it: rather than using a `match`, we could just use the `?` operator here to propagate the error from `s`:
```suggestion
                .map(move |s|  {
                    let (_, conn) = s?.get_ref();
                    let end_cert = extract_cert(conn)?;
                    verify::verify_id(end_cert, &server_id)?;
                    Ok(ClientIo(s))
                }),
```